libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/NavigationView.java

```diff
@@ -65,7 +66,7 @@
  *
  * @since 0.7.0
  */
-public class NavigationView extends CoordinatorLayout implements LifecycleObserver, OnMapReadyCallback,
+public class NavigationView extends CoordinatorLayout implements LifecycleOwner, OnMapReadyCallback,
```

Comment:
since every Activity and Fragment implements `LifecycleOwner` i'm not sure if this is a legit fix, considering that we already expose `on*` callbacks would it be reasonable to expect users to wire them up to activity or fragment lifecycle callbacks?

Reply:
That is exactly why I did it. When we implement `LifecycleOwner` in `NavigationView` - we remove the obligation from user to pass LifecycleOwner into this view by smth like `setLifecycleOwner()`. In our [documentation](https://docs.mapbox.com/android/navigation/examples/navigationview-with-fragment/) we describes how to operate with `NavigationView`, and there only need to call `on*` callbacks. All other stuff regarding lifecycle in `NavigationView` we made internally and we shouldn't ask users to do any other movements. So implementing `NavigationView` as `LifecycleOwner` (for it "child" views, like `IntructionView`, `SummaryBottomSheet` and so on) allow us to minimize additional user interactions with our `NavigationView` for it correct work. 
I mean `NavigationView` is smth like "parent" or container for other views (`IntructionView`, `SummaryBottomSheet`, etc.), so in this nested views we just subscribe for observing lifecycle of its "parent" (`NavigationView`).

Reply:
so afaik view has much simpler lifecycle, looking at the change are we essentially forcing activity/fragment lifecycle onto view? what's the benefit of doing it? 
cc: @Guardiola31337

Reply:
in my perspective this view has grown beyond the scope of what any view should contain. leading to a fragment would be a logical choice. but then again there are people that use views for everything and completely avoid fragments and activities so this seems like a personal choice more than anything